Allow users to restrict media type on Fieldmanager_Media

firejdl opened the pull request:
This commit adds a "media_type" parameter on the Fieldmanager_Media object, which allows developers to restrict the media type for the Media Library that is opened. Valid options are "all", "image", "audio", or "video".

mboynes commented:
Turns out this PR is more capable than you realize. It looks like you can use any mime type like "application/pdf", or the first part of any mime type (before the slash), like "application". That's awesome.

Can you...
* [ ] Update the inline docs to cover this
* [ ] Merge in the latest master (that should fix the failing unit tests)
* [ ] Change `"media_type"` to `"mime_type"`
* [ ] Add a basic unit test (just verify that when adding `mime_type` that the `data-type` attribute is successfully added)

firejdl commented:
Hi @mboynes:

I updated the pull request with the changes you requested, and also fixed a bug that we found when the user can add multiple items (`$limit` > 1).

mboynes commented:
Replaced by #416
